test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/survival/DbSurvivalIntegrationTest.java

```diff
@@ -0,0 +1,126 @@
+    }
+
+    @Before
+    public void setUp() {
+        WorkbenchUtils.deployProjectToWorkbench(gitProvider, deploymentScenario.getWorkbenchDeployment(), DEFINITION_PROJECT_NAME);
+
+        kieServerMgmtControllerClient = KieServerControllerClientProvider.getKieServerMgmtControllerClient(deploymentScenario.getWorkbenchDeployment());
+
+        kieServicesClient = KieServerClientProvider.getKieServerClient(deploymentScenario.getKieServerDeployment());
+        processServicesClient = KieServerClientProvider.getProcessClient(deploymentScenario.getKieServerDeployment());
+        queryServicesClient = KieServerClientProvider.getQueryClient(deploymentScenario.getKieServerDeployment());
+    }
+
+    @Test
+    public void reconnectionDbTest() {
+        logger.debug("Register Kie Container to Kie Server");
+        KieServerInfo serverInfo = kieServicesClient.getServerInfo().getResult();
+        kieServerMgmtControllerClient.saveContainerSpec(serverInfo.getServerId(), serverInfo.getName(), CONTAINER_ID, CONTAINER_ALIAS, PROJECT_GROUP_ID, DEFINITION_PROJECT_NAME, DEFINITION_PROJECT_VERSION, KieContainerStatus.STARTED);
+        KieServerClientProvider.waitForContainerStart(deploymentScenario.getKieServerDeployment(), CONTAINER_ID);
+
+        logger.debug("Start process instance");
+        Long signalPid = processServicesClient.startProcess(CONTAINER_ID, SIGNALTASK_PROCESS_ID);
+        assertThat(signalPid).isNotNull().isGreaterThan(0L);
+        assertThat(queryServicesClient.findProcessInstances(0, 10)).isNotNull().hasSize(1);
+
+        scaleDatabaseTo(0);
+
+        logger.debug("Try to get process instances");
+        assertThatThrownBy(() -> {
+            processServicesClient.startProcess(CONTAINER_ID, SIGNALTASK_PROCESS_ID);
+        }).isInstanceOf(KieServicesException.class);
+
+        scaleDatabaseTo(1);
+
+        waitForKieServerResposne();
+
+        logger.debug("Check started processes");
+        assertThat(queryServicesClient.findProcessInstances(0, 10)).isNotNull().hasSize(1);
+
+        logger.debug("Send signal and complete process");
+        processServicesClient.signal(CONTAINER_ID, SIGNAL_NAME, null);
+
+        logger.debug("Check that prcoess is completed");
+        assertThat(queryServicesClient.findProcessInstanceById(signalPid).getState()).isEqualTo(ProcessInstance.STATE_COMPLETED);
+
+    }
+
+    private void scaleDatabaseTo(int count) {
+        logger.debug("Scale Database to " + count);
+        deploymentScenario.getDatabaseDeployment().scale(count);
+        deploymentScenario.getDatabaseDeployment().waitForScale();
+    }
+
+    private void waitForKieServerResposne() {
+        LocalDateTime endTime = LocalDateTime.now().plusSeconds(30);
+
+        while (LocalDateTime.now().isBefore(endTime)) {
+            try {
+                assertThat(queryServicesClient.findProcessInstances(0, 10)).isNotNull();
+                return;
+            } catch (Exception e) {
```

Comment:
Is there just a KieServicesException expected? Please catch just that one, so it doesn't hide any other issue.